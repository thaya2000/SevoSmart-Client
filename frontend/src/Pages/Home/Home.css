
.Home{
    display: flex;
    flex-direction: column;
    
    width: 100%;
    
   
}
@media screen and (min-width: 900px) {
    .Home{

    
    width: 110%;
    
   
}
}


