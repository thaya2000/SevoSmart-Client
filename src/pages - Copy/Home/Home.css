.home {
  position: relative;
  overflow-y: auto;
  scroll-snap-type: y mandatory;
  height: 100vh;
  width: 100%;
  -ms-overflow-style: none;
}

.home::-webkit-scrollbar {
  display: none;
}
