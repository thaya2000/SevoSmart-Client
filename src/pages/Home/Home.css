.home {
  position: relative;
  overflow-y: auto;
  overflow-x: hidden;
  scroll-snap-type: y mandatory;
  height: 100%;
  width: 100%;
  -ms-overflow-style: none;
}

.home::-webkit-scrollbar {
  display: none;
}

.intro {
  height: 90vh;
  width: 100%;
  display: flex;
  background-color: #040e42;
}

.introImage {
  display: flex;
  height: 90vh;
}

@keyframes blink-scale {
  0% {
    opacity: 1;
    transform: scale(1);
  }
  50% {
    opacity: 0;
    transform: scale(1.2);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
}
