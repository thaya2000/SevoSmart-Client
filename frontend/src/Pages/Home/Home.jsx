import React, { useRef, useState, useEffect } from "react";
import { debounce } from "lodash";
import Navigation from "../../Components/Navigation/Navigation.jsx";
import ContentSolar from "../../Components/Content/ContentSolar";
import ContentConstructions from "../../Components/Content/ContentConstructions";
import Footer from "../../Components/Footer/Footer";
import "./Home.css";
import Content from "../../Components/Content/Content";

const Home = () => {
  const contentRefs = [useRef(null), useRef(null), useRef(null)];
  const [activeSlide, setActiveSlide] = useState(0);

  const handleWheelScroll = debounce((event) => {
    const deltaY = event.deltaY;
    const scrollDelta = deltaY > 0 ? 1 : -1;
    const newSlide = Math.max(0, Math.min(activeSlide + scrollDelta, 2));

    if (newSlide !== activeSlide) {
      setActiveSlide(newSlide);

      // Manually scroll to the new slide position
      const targetPosition = contentRefs[newSlide].current.offsetTop;

      // Use smooth scroll to the target position
      smoothScrollTo(targetPosition);
    }
  }, 800); // Adjust the debounce time as needed

  const smoothScrollTo = (targetPosition) => {
    const currentPosition = window.scrollY;
    const distance = targetPosition - currentPosition;

    const startTime = performance.now();
    const duration = 800; // Adjust as needed

    const animateScroll = (currentTime) => {
      const elapsedTime = currentTime - startTime;
      const progress = Math.min(elapsedTime / duration, 1);
      const easing = easeInOutQuad(progress);
      window.scrollTo(0, currentPosition + distance * easing);

      if (progress < 1) {
        requestAnimationFrame(animateScroll);
      }
    };

    requestAnimationFrame(animateScroll);
  };

  const easeInOutQuad = (t) => (t < 0.5 ? 2 * t * t : -1 + (4 - 2 * t) * t);

  useEffect(() => {
    const handleScroll = () => {
      const currentSlide = Math.floor(window.scrollY / window.innerHeight);
      setActiveSlide(currentSlide);
    };

    window.addEventListener("scroll", handleScroll);

    return () => {
      window.removeEventListener("scroll", handleScroll);
    };
  }, []);

  return (
    <div className="Home" onWheel={handleWheelScroll}>
      <Navigation />

      <div className="slides-container">
        <div ref={contentRefs[0]}>
          <Content />
        </div>
        <div ref={contentRefs[1]}>
          <ContentSolar />
        </div>
        <div ref={contentRefs[2]}>
          <ContentConstructions />
        </div>
        <div>
          <Footer />
        </div>
      </div>
    </div>
  );
};

export default Home;
