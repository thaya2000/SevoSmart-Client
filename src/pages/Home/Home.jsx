import { useState, useEffect } from "react";
import "./Home.css";
import IntroText from "../../component/HomeComponent/IntroText/IntroText.jsx";
import ProductIntroCard from "../../component/HomeComponent/ProductIntroCard/ProductIntroCard.jsx";
import SolarCoverImage from "../../assets/solar.jpg";
import ConstructionCoverImage from "../../assets/construction.jpg";
import image1 from "../../Images/cons1.jpg";
import image2 from "../../Images/pro1.jpg";
import image3 from "../../Images/S1.jpeg";
import IntroImageSlider from "../../component/HomeComponent/IntroImageSlider/IntroImageSlider.jsx";
import PastProject from "../../component/HomeComponent/PastProject/PastProject.jsx";
import BillCalculator from "../../component/HomeComponent/BillCalculator/BillCalculator.jsx";
import OfferPanel from "../../component/HomeComponent/OfferPanel/OfferPanel/OfferPanel.jsx";
import axios from "axios";
import RambousLoader from "../../routes/RambousLoader";

const Home = () => {
  const [pastProjects, setPastProjects] = useState([]);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    const fetchPastProjects = async () => {
      try {
        setLoading(true);
        const result = await axios.get("/api/v1/admin/past-projects");
        setLoading(false);
        console.log(result);

        // Transform the data to match the structure required by your component
        const transformedData = result.data.map((project) => ({
          name: project.projectName,
          // image: project.productImageURL[0], // Taking the first image
          image: project.productImageURL[0], // Taking the first image
          description: project.description,
        }));

        setPastProjects(transformedData);
      } catch (error) {
        console.error("Error fetching past projects:", error);
      }
    };

    fetchPastProjects();
  }, []);

  const images = [image1, image2, image3];

  return (
    <div>
      {loading ? (
        <RambousLoader />
      ) : (
        <div className="home">
          <div className="intro grid grid-cols-1 xl:grid-cols-2">
            <div className="grid sm:grid-rows-2">
              <div className="flex">
                <IntroText />
              </div>
              <div className="flex justify-center items-center w-full">
                <OfferPanel />
              </div>
            </div>
            <div className="introImage w-full">
              <IntroImageSlider images={images} />
            </div>
          </div>
          <div className="flex w-full relative">
            <PastProject pastProjects={pastProjects} />
          </div>
          <ProductIntroCard
            image={SolarCoverImage}
            serviceTitle="Solar Panels"
            orderLink="/energy-order"
            learnMoreLink="/energy-learnmore"
            buttonText="Order Now"
          />
          <ProductIntroCard
            image={ConstructionCoverImage}
            serviceTitle="Constructions"
            orderLink="/new-building-consultation"
            learnMoreLink="/construction-learnmore"
            textColor="#ffffff"
            buttonText="Consult Now"
          />
          <div>
            <BillCalculator />
          </div>
        </div>
      )}
    </div>
  );
};

export default Home;
