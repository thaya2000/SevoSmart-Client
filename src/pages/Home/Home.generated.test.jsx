import { describe, it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import Home from './Home';

vi.mock("./Home.css");
vi.mock("../../component/HomeComponent/IntroText/IntroText.jsx");
vi.mock("../../component/HomeComponent/ProductIntroCard/ProductIntroCard.jsx");
vi.mock("../../assets/solar.jpg");
vi.mock("../../assets/construction.jpg");
vi.mock("../../assets/footer_sample.jpg");
vi.mock("../../component/HomeComponent/IntroImageSlider/IntroImageSlider.jsx");
vi.mock("../../component/HomeComponent/PastProject/PastProject.jsx");
vi.mock("../../component/HomeComponent/BillCalculator/BillCalculator.jsx");
vi.mock("../../component/HomeComponent/OfferPanel/OfferPanel/OfferPanel.jsx");

describe('<Home>', () => {
  it('should render component', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
