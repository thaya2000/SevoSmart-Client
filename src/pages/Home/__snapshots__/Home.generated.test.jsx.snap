// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Home> > should render component 1`] = `
<DocumentFragment>
  <div
    class="home "
  >
    <div
      class="intro grid grid-cols-1 xl:grid-cols-2"
    >
      <div
        class="grid sm:grid-rows-2"
      >
        <div
          class="flex "
        />
        <div
          class="flex justify-center items-center w-full"
        />
      </div>
      <div
        class="introImage w-full"
      />
    </div>
    <div
      class="flex w-full relative"
    />
    <div />
  </div>
</DocumentFragment>
`;
